assets/prototype/domain/shared/predicates/dateUtilities/add-subtract.ts

```diff
@@ -0,0 +1,53 @@
+import {
+	addDays,
+	addHours,
+	addISOWeekYears,
+	addMilliseconds,
+	addMinutes,
+	addMonths,
+	addQuarters,
+	addSeconds,
+	addWeeks,
+	addYears,
+	toDate,
+} from 'date-fns';
+
+import { toInteger } from '../../../../application/utilities/converters/number';
+
+type IntervalType =
+	| 'days'
+	| 'hours'
+	| 'ISOWeekYears'
+	| 'milliseconds'
+	| 'minutes'
+	| 'months'
+	| 'quarters'
+	| 'seconds'
+	| 'weeks'
+	| 'years';
+
+const addMapping = {
+	days: addDays,
+	hours: addHours,
+	ISOWeekYears: addISOWeekYears,
+	milliseconds: addMilliseconds,
+	minutes: addMinutes,
+	months: addMonths,
+	quarters: addQuarters,
+	seconds: addSeconds,
+	weeks: addWeeks,
+	years: addYears,
+};
+
+export const add = (interval: IntervalType, date: Date, amount: number) => {
+	const func = addMapping[interval];
+	return func(date, amount);
+};
+
+export const sub = (interval: IntervalType, dirtyDate: Date, dirtyAmount: number) => {
+	const func = addMapping[interval];
+	const amount = toInteger(dirtyAmount);
+	const date = toDate(dirtyDate);
+
+	return func(date, -amount);
```

Comment:
I've used the same approach as in `date-fns`' sub* op: https://github.com/date-fns/date-fns/blob/master/src/subMinutes/index.js#L34